spec.html

```diff
@@ -21150,22 +21151,19 @@ <h1>Runtime Semantics: ClassDefinitionEvaluation</h1>
         1. Let _proto_ be ! OrdinaryObjectCreate(_protoParent_).
         1. If |ClassBody_opt| is not present, let _constructor_ be ~empty~.
         1. Else, let _constructor_ be ConstructorMethod of |ClassBody|.
-        1. If _constructor_ is ~empty~, then
-          1. If |ClassHeritage_opt| is present, then
-            1. Set _constructor_ to the result of parsing the source text
-              <pre><code class="javascript">constructor(...args) { super(...args); }</code></pre>
-              using the syntactic grammar with the goal symbol |MethodDefinition[~Yield, ~Await]|.
-          1. Else,
-            1. Set _constructor_ to the result of parsing the source text
-              <pre><code class="javascript">constructor() {}</code></pre>
-              using the syntactic grammar with the goal symbol |MethodDefinition[~Yield, ~Await]|.
         1. Set the running execution context's LexicalEnvironment to _classScope_.
-        1. Let _constructorInfo_ be ! DefineMethod of _constructor_ with arguments _proto_ and _constructorParent_.
-        1. Let _F_ be _constructorInfo_.[[Closure]].
+        1. If _constructor_ is ~empty~, then
+          1. Let _steps_ be the algorithm steps defined in <emu-xref href="#sec-default-constructor-functions"></emu-xref>.
+          1. Let _F_ be ! CreateBuiltinFunction(_steps_, &laquo; [[ConstructorKind]], [[HomeObject]], [[SourceText]] &raquo;, ~empty~, _constructorParent_, *true*).
+          1. Perform ! SetFunctionLength(_F_, *+0*<sub>𝔽</sub>).
+          1. Set _F_.[[HomeObject]] to _proto_.
```

Comment:
Note that this branch also needs to set <code><var>F</var>.[[SourceText]]</code>.

Reply:
It doesn't, that's set by the caller to ClassDefinitionEvaluation.